go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -179,51 +253,70 @@ func createWorkingSetForLocalBranch(ctx *sql.Context, ddb *doltdb.DoltDB, branch
 
 // checkoutRemoteBranch checks out a remote branch creating a new local branch with the same name as the remote branch
 // and set its upstream. The upstream persists out of sql session.
-func checkoutRemoteBranch(ctx *sql.Context, dbName string, dbData env.DbData, branchName string, apr *argparser.ArgParseResults, rsc *doltdb.ReplicationStatusController) error {
+func checkoutRemoteBranch(ctx *sql.Context, dSess *dsess.DoltSession, dbName string, dbData env.DbData, branchName string, apr *argparser.ArgParseResults, rsc *doltdb.ReplicationStatusController) (string, error) {
 	remoteRefs, err := actions.GetRemoteBranchRef(ctx, dbData.Ddb, branchName)
 	if err != nil {
-		return errors.New("fatal: unable to read from data repository")
+		return "", errors.New("fatal: unable to read from data repository")
 	}
 
 	if len(remoteRefs) == 0 {
-		return fmt.Errorf("error: could not find %s", branchName)
+		if doltdb.IsValidCommitHash(branchName) && apr.Contains(cli.MoveFlag) {
+
+			// User tried to enter a detached head state, which we don't support.
+			// Inform and suggest that they check-out a new branch at this commit instead.
+
+			return "", fmt.Errorf("dolt does not support a detached head state. To create a branch at this commit instead, run:\n\n"+
```

Comment:
Consider backticks for multi-line output. It's more readable. Non-blocking

Reply:
Done.